### DISCUSSION TITLE
KDE/Plasma  Ethernet connection not seen

### DISCUSSION THREAD

**Discussion by gitJWR** [SpiralLinux/SpiralLinux-project#176] [Q&A]
Hello,
I have been using the Gnome edition, but decided to switch to Plasma this afternoon.  Fresh install seemed to go well, but I am now noticing my ethernet connection does not seem to be recognized by the network manager.  It does see my wi-fi network, and I can connect to that, but would prefer my ethernet connection.  Any idea what's going on?
Thanks

**Comment by geckolinux** [comment]
Hi there, any idea what chipset your ethernet uses? Did you try with both the 5.10 and the 5.18 kernel?

**Comment by gitJWR** [comment]
![Screenshot_20221027_211520](https://user-images.githubusercontent.com/20615106/198465817-1955d93a-09a2-413d-9758-427f94112dfe.png)
Sorry...not sure where to find chipset info.  Looks like update after install has me on 5.19 kernel.  Have not tried 5.10, but would need a little guidance as to how I switch.  Not sure if any of this neofetch output is relevant.

**Reply by geckolinux** [reply]
Hi there, the other kernel is available in the GRUB boot menu.

**Reply by gitJWR** [reply]
Thank you!  On kernel 5.10, ethernet was immediately found and connected.

**Reply by geckolinux** [reply]
Interesting, thanks a lot for confirming that. I'd like to know what the chipset and driver are. Could you please post the output of:
`sudo inxi -N`

**Reply by gitJWR** [reply]
of course!  I will post that output tonight, when I'm back home.  (do I just copy/paste, or is there another preferred method?)

**Reply by gitJWR** [reply]
Another related question for you.  Can you tell me how to persistently boot into that 5.10 kernel?  I went into the Grub Customizer tool and moved it to the top of the list...but it still defaults to booting 5.19, so clearly I am missing something.  Thanks for all of your help!

**Reply by gitJWR** [reply]
> Interesting, thanks a lot for confirming that. I'd like to know what the chipset and driver are. Could you please post the output of: `sudo inxi -N`

Network:   Device-1: Qualcomm Atheros QCA6174 802.11ac Wireless Network Adapter driver: ath10k_pci 
           Device-2: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet driver: r8168 
           Device-3: Qualcomm Atheros type: USB driver: btusb

**Comment by artytux** [comment]
Hello
I did my updates a day ago
firmware-realtek (20210315-3) to 20210818-1~bpo11+1     <-<-Could this be the problem

It's the only part of updates that seems to point to connectivity
I was on LAN for years first time problem with LAN
ran this and results with

fa@fasystem:~$ sudo rfkill list all
[sudo] password for fa:         
0: phy0: Wireless LAN
        Soft blocked: no
        Hard blocked: no
1: hci0: Bluetooth
        Soft blocked: no
        Hard blocked: no

then
fa@fasystem:~$ inxi -N
Network:   Device-1: Realtek RTL8192EE PCIe Wireless Network Adapter driver: rtl8192ee 
           Device-2: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet driver: N/A 

Funny thing it was there working getting the update,
fortunately SpiralLinux has wireless now that's to be thankful for, Phew.

PCI Express Gigabit Ethernet driver: N/A

I won't be back until 20 Hrs time approx, hope this helps a little bit, and I'll be able to help more then, Have To have to finish off $$ job.

Regards
artytux

**Reply by CicadaSeventeen** [reply]
No, not about firmware I guess. Please tell me if your kernel is 5.19. It is easy to understand if `r8168` not wors on kernel 5.19

**Reply by CicadaSeventeen** [reply]
see https://github.com/SpiralLinux/SpiralLinux-project/issues/175

**Reply by CicadaSeventeen** [reply]
You can just remove `r8168-dkms` and test if it works.

**Comment by artytux** [comment]
Hello
@CicadaSeventeen 
Completely forgot to mention that, Apologizes,
I _am_ on **kernel 5.19**

When rebooted yesterday ?  with 5.18 do have LAN working
not on 5.19 kernel
Removal of r8168-dkms on 5.19 still has no Lan waited 3.30 minutes usually Lan kicks in in about 2 minutes, so no change, worth a try,

I will wait for your reply before replacing  r8168-dkms incase you want to try something else

So is this a case of Wait for Debian to issue a patch and keep using WiFi until . . . 

Regards
artytux

**Reply by CicadaSeventeen** [reply]
> Removal of r8168-dkms on 5.19 still has no Lan waited 3.30 minutes usually Lan kicks in in about 2 minutes, so no change, worth a try,
Oh I forgot to tell you. You have to remove `r8168-dkms` package and reboot to find out if Wlan can work.
Besides, I am not English native speaker, so please write more clearly otherwise I can get confused with your words.

**Comment by artytux** [comment]
Hello
@CicadaSeventeen 
I am on kernel 5.19
Removed r8168-dkms and
did reboot (I knew that Thanks)

LAN still not work . . . only WiFi works.


Regards
artytux

**Reply by CicadaSeventeen** [reply]
See [here](https://github.com/SpiralLinux/SpiralLinux-project/issues/175), try the package I provide. This is the last thing I know.

**Comment by artytux** [comment]
Hello
@CicadaSeventeen 
Well Good installed package [r8168-temp.tar.gz] and yes rebooted and then Lan works OK, WiFi works OK,

For anyone having same problems please post _**if not solution**_

Have a good day

Thank  You heaps

@gitJWR, Have you tried this and does this make Lan work for you, IF you are not sure about just download from link and left click on deb file in downloads and it will install itself, get problems please post.

Regards
artytux

**Reply by geckolinux** [reply]
Thanks very much to @CicadaSeventeen for the fixed package. It looks like the `r8168-dkms` package is going to be problematic, and thanks to @CicadaSeventeen for noting that it's not really needed for most Realtek 8168 hardware that is covered by the official in-tree r8169 driver:
https://packages.debian.org/bullseye-backports/r8168-dkms
> This driver should only be used for devices not yet supported by the in-kernel driver r8169.
